### PR TITLE
Support over 256 macro arguments

### PR DESCRIPTION
**Pull request by aaaaaa123456789**
This removes the limit that was created by a fixed array, boosting the maximum to whatever can be held in memory. Currently a maximum of 99,999 is enforced.

### CODE REVIEW

**Comment by JL2210** [comment]
Woohoo! Next step is removing the max and just setting the limit as "whatever can be stored in memory".

**Comment by aaaaaa123456789** [comment]
I left the max in because you might want to enforce _some_ limit, but you can make it customizable.

**Comment by JL2210** [comment]
I don't see why someone would want to enforce some limit. And even if they did and they wanted to customize it, they'd have to recompile to get any effect.

**Comment by ISSOtm** [comment]
The difference is that the changes made make it almost trivial to make the limit fixed at run-time; the only compile-time dependency left is the "max" error message. This can easily be changed, however.

Also, a limit may make sense wrt memory consumption.

File: src/asm/macro.c [hunk 1]
```diff
@@ -40,12 +39,13 @@ struct MacroArgs *macro_NewArgs(void)
...
 
-void macro_AppendArg(struct MacroArgs *args, char *s)
+void macro_AppendArg(struct MacroArgs **args, char *s)
 {
-	if (args->nbArgs == MAXMACROARGS)
+	if ((**args).nbArgs == MAXMACROARGS)
 		yyerror("A maximum of " EXPAND_AND_STR(MAXMACROARGS)
 			" arguments is allowed");
-	args->args[args->nbArgs++] = s;
+	*args = realloc(*args, sizeof **args + sizeof (char *) * (1 + (**args).nbArgs));
```
Comment (NieDzejkob) [inline src/asm/macro.c]: Note that this will result in the execution time being quadratic in number of simultaneous macro arguments.

Comment (aaaaaa123456789) [reply src/asm/macro.c]: Only if the reallocation doesn't happen in place, which is unlikely.

Comment (NieDzejkob) [reply src/asm/macro.c]: That depends on how fragmented the heap is. By my measurements, the buffer still needs to be reallocated about 20% of the time when assembling a big project.

**Pull request merged**